devicetypes/smartthings/fibaro-heat-controller.src/fibaro-heat-controller.groovy

```diff
@@ -0,0 +1,334 @@
+	sendEventToChild(map)
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	if(cmd.parameterNumber == 3 && cmd.scaledConfigurationValue == 1) {
+		if(!childDevices) {
+			addChild()
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationBusy cmd) {
+	log.warn "Device is busy, delaying refresh"
+	runIn(15, "forcedRefresh", [overwrite: true])
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.warn "Unhandled command: ${cmd}"
+	[:]
+}
+
+def setThermostatMode(String mode) {
+	def modeValue = 0
+	switch (mode) {
+		case "heat":
+			modeValue = 1
+			break
+		case "emergency heat":
+			modeValue = 31
+			break
+		case "off":
+			modeValue = 0
+			break
+	}
+
+	[
+			secureEncap(zwave.thermostatModeV2.thermostatModeSet(mode: modeValue), 1),
+			"delay 2000",
+			secureEncap(zwave.thermostatModeV2.thermostatModeGet(), 1)
+	]
+}
+
+def heat() {
+	setThermostatMode("heat")
+}
+
+def off() {
+	setThermostatMode("off")
+}
+
+def emergencyHeat() {
+	setThermostatMode("emergency heat")
+}
+
+def setHeatingSetpoint(setpoint) {
+	setpoint = temperatureScale == 'C' ? setpoint : fahrenheitToCelsius(setpoint)
+	[
+			secureEncap(zwave.thermostatSetpointV2.thermostatSetpointSet([precision: 1, scale: 0, scaledValue: setpoint, setpointType: 1, size: 2]), 1),
+			"delay 2000",
+			secureEncap(zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: 1), 1)
+	]
+}
+
+def setThermostatSetpointUp() {
+	def setpoint = device.latestValue("heatingSetpoint")
+	if (setpoint < maxHeatingSetpointTemperature) {
+		setpoint = setpoint + (temperatureScale == 'C' ? 0.5 : 1)
+	}
+	setHeatingSetpoint(setpoint)
+}
+
+def setThermostatSetpointDown() {
+	def setpoint = device.latestValue("heatingSetpoint")
+	if (setpoint > minHeatingSetpointTemperature) {
+		setpoint = setpoint - (temperatureScale == 'C' ? 0.5 : 1)
+	}
+	setHeatingSetpoint(setpoint)
+}
+
+def refresh() {
+	def cmds = [
+			secureEncap(zwave.batteryV1.batteryGet(), 1),
+			secureEncap(zwave.batteryV1.batteryGet(), 2),
+			secureEncap(zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: 1)),
+			secureEncap(zwave.thermostatModeV2.thermostatModeGet()),
+			secureEncap(zwave.sensorMultilevelV5.sensorMultilevelGet()),
+			secureEncap(zwave.configurationV1.configurationGet(parameterNumber: 3))
+	]
+
+	delayBetween(cmds, 2500)
+}
+
+def ping() {
+	refresh()
+}
+
+private secureEncap(cmd, endpoint = null) {
+	secure(encap(cmd, endpoint))
+}
+
+private secure(cmd) {
+	if(zwaveInfo.zw.endsWith("s")) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+def switchMode() {
```

Comment:
Do we ever plan to support updating this list dynamically? It seems like right now it is a fixed list, which means that we could, in theory, just have the action for each state in the tiles above call what would normally be next in the list. Off calls heat(), heat calls emergencyHeat(), emergency heat calls off().

Also, is there ever a time where supportedModes[0].size() would be 1 character long or less?

Reply:
Generally, I used it when I wasn't sure about how to handle additional temperature sensor. For now, there's no need for dynamically updated list. 
And no, in current state there's no chance that supportedModes[0].size() would be 1 char or less.